frontend/src/components/elements/VegaLiteChart/VegaLiteChart.tsx

```diff
@@ -252,15 +246,19 @@ class VegaLiteChart extends React.PureComponent<Props, State> {
 
     const spec = JSON.parse(el.get("spec"))
 
-    if (spec.width === 0) {
-      spec.width = this.props.width - EMBED_PADDING
+    const { width, height } = this.getChartDimensions()
+    spec.width = width
+    spec.height = height
+    spec.padding = {
+      bottom: EMBED_PADDING,
```

Comment:
Wait, this will override any padding the user adds, right?

Reply:
Yes. Can they set a padding when write the chart? I that case, I can use that padding and this should be the default one.